docker_images/diffusers/app/pipelines/image_to_image.py

```diff
@@ -167,15 +176,46 @@ def _process_req(self, image, prompt, **kwargs):
                 StableDiffusionDepth2ImgPipeline,
             ),
         ):
+            if "num_inference_steps" not in kwargs:
+                kwargs["num_inference_steps"] = 25
             images = self.ldm(prompt, image, **kwargs)["images"]
             return images[0]
         elif isinstance(self.ldm, (StableUnCLIPImg2ImgPipeline, StableUnCLIPPipeline)):
+            if "num_inference_steps" not in kwargs:
+                kwargs["num_inference_steps"] = 25
             # image comes first
             images = self.ldm(image, prompt, **kwargs)["images"]
             return images[0]
         elif isinstance(self.ldm, StableDiffusionImageVariationPipeline):
+            if "num_inference_steps" not in kwargs:
+                kwargs["num_inference_steps"] = 25
             # only image is needed
             images = self.ldm(image, **kwargs)["images"]
             return images[0]
+        elif isinstance(self.ldm, (KandinskyImg2ImgPipeline)):
+            if "num_inference_steps" not in kwargs:
+                kwargs["num_inference_steps"] = 100
+            # not all args are supported by the prior
+            prior_args = {
+                "num_inference_steps": kwargs["num_inference_steps"],
+                "num_images_per_prompt": kwargs["num_images_per_prompt"],
+                "negative_prompt": kwargs.get("negative_prompt", None),
+                "guidance_scale": kwargs.get("guidance_scale", 7),
+            }
+            prior_emb = self.prior(prompt, **prior_args)
+            image_emb = prior_emb.images
+            zero_image_emb = prior_emb.zero_embeds
+            if "negative_prompt" in kwargs:
+                zero_image_emb = self.prior(
+                    kwargs["negative_prompt"], **prior_args
+                ).images
```

Comment:
@yiyixuxu please let me know if this make sense, if user doesn't send a `negative_prompt` then we run prior with  the `prompt` and `negative_prompt=none`, get `image_emb`  and `zero_image_emb`, otherwise run prior again replacing `zero_image_emb` is that correct?

Reply:
Once https://github.com/huggingface/diffusers/pull/3636 is merged we can remove this

Reply:
Once https://github.com/huggingface/diffusers/pull/3636 is merged we can remove this
```suggestion
            prior_emb = self.prior(prompt, **prior_args)
            image_emb = prior_emb.images
            zero_image_emb = prior_emb.zero_embeds
```